File: src/services.js
/**
 * Service definitions for critical infrastructure ports.
 *
 * Each service entry maps a human-readable name to its expected port,
 * host, health endpoint, and a brief description of what it provides.
 */

'use strict';

const SERVICES = Object.freeze([
  { name: 'oscar',   port: 11434, host: 'localhost', description: 'Ollama LLM inference service' },
  { name: 'honcho',  port: 8000,  host: 'localhost', description: 'Honcho API gateway / orchestration service' },
  { name: 'huly',    port: 8087,  host: 'localhost', description: 'Huly project management platform' },
  { name: 'phantom', port: 3100,  host: 'localhost', description: 'Phantom headless rendering service' },
]);

module.exports = { SERVICES };

File: scripts/healthcheck/services.py
"""Service definitions for critical infrastructure ports."""

from dataclasses import dataclass


@dataclass(frozen=True)
class Service:
    """Represents an infrastructure service with its health check config."""

    name: str
    port: int
    host: str = "localhost"
    protocol: str = "tcp"
    description: str = ""
    critical: bool = True


# Registry of critical services that must be running.
CRITICAL_SERVICES: list[Service] = [
    Service(
        name="oscar",
        port=11434,
        description="Oscar AI service",
    ),
    Service(
        name="honcho",
        port=8000,
        description="Honcho orchestration service",
    ),
    Service(
        name="huly",
        port=8087,
        description="Huly project management service",
    ),
    Service(
        name="phantom",
        port=3100,
        description="Phantom rendering service",
    ),
]
